server/pbenchinacan/load_keycloak.sh

```diff
@@ -17,13 +17,50 @@
 # "http://localhost:8080/*" unless specified otherwise by 'KEYCLOAK_REDIRECT_URI'
 # env variable.
 
+KEYCLOAK_BASE_IMAGE=${KEYCLOAK_BASE_IMAGE:-"images.paas.redhat.com/pbench/pbenchinacan-keycloak:base"}
 KEYCLOAK_HOST_PORT=${KEYCLOAK_HOST_PORT:-"http://localhost:8090"}
+KEYCLOAK_HEALTH_URI="${KEYCLOAK_HOST_PORT}/health"
 KEYCLOAK_REDIRECT_URI=${KEYCLOAK_REDIRECT_URI:-"http://localhost:8080/*"}
 ADMIN_USERNAME=${ADMIN_USERNAME:-"admin"}
 ADMIN_PASSWORD=${ADMIN_PASSWORD:-"admin"}
 REALM=${KEYCLOAK_REALM:-"pbench"}
 CLIENT=${KEYCLOAK_CLIENT:-"pbench-server"}
 
+podman run -d --rm -p 8090:8090 \
+-e KEYCLOAK_ADMIN=admin -e KEYCLOAK_ADMIN_PASSWORD=admin \
+${KEYCLOAK_BASE_IMAGE} start-dev --health-enabled=true --http-port=8090
```

Comment:
Based on our discussion yesterday, the intent of this script is now to target an already-running Keycloak instance -- in the case of the functional tests, it will have been started previously by an outer script (e.g., by the `pod.yml` file or the equivalent individual `podman` commands), and, in the case of the Staging server, it will have been started by the person doing the deployment.

Therefore, _this_ script shouldn't be running any Keycloak container.  (And, therefore, you don't need to define `KEYCLOAK_BASE_IMAGE`.)

Reply:
Please take a look at PR #3216, I am removing this command from the commit in that PR.

Reply:
Given that this script isn't currently executed by the CI or any other part of the Pbench infrastructure, it's not obvious to me why it matters whether the `podman` command is added here or not, _particularly_ if the next PR is going to remove it again!  (Why not omit the add here and then we don't have to worry about removing it?)

Or, as I [suggested](https://github.com/distributed-system-analysis/pbench/pull/3216#discussion_r1090848707) in #3216, perhaps you should merge these two PRs.

Reply:
> Given that this script isn't currently executed by the CI or any other part of the Pbench infrastructure, it's not obvious to me why it matters whether the `podman` command is added here or not, _particularly_ if the next PR is going to remove it again! (Why not omit the add here and then we don't have to worry about removing it?)

My reasoning is that I wanted to make changes incrementally. So we can go back in our commit history on main and see how the file has evolved with the related Jira issues. Indeed it is not executed by the CI script at the time of this PR but removing the podman run command is really a part of an issue PR #3216 is trying to solve. If I make that change here because eventually, we are anyway going to remove it then I feel we lose that linear history of the file that we can go back to in our git commit history, isn't it? 


> Or, as I [suggested](https://github.com/distributed-system-analysis/pbench/pull/3216#discussion_r1090848707) in #3216, perhaps you should merge these two PRs.

Yeah, perhaps I could keep 2 commits with separate jira issue numbers and but #3216 is already rebased on top of this PR so I don't know if that's worth it.

Reply:
The historical perspective looks back along the sequence of commits to the `main` branch, not to the PR branches (those are typically deleted after they are merged, and they actually come from different (forked) repos anyway, so we cannot depend on them persisting after they are merged).

From that perspective, this PR _adds_ the `podman` command which the next one will remove again...that's not really an increment so much as a tangential side-trip.

What's more significant is that you've already removed from this script the code which _shuts down_ the container, so having the script start the container potentially leaves it hanging around after the script is done.  (I presume that the intention is to dump onto the user the responsibility of persisting the container image and cleaning up, but, I as a user, would prefer the script to do that for me!  And, it's a model that we've already agreed to move away from.)

And, the script as it stands is useless in the Staging environment...whereas, without the `podman` command, it would be immediately useful (when it's not being used in any other way).

So, I really don't see any reason why we should merge this PR which is adding something that we don't want and which we are just about to tear out again.  🤷

Reply:
I agree with @webbnh here.

Reply:
Okay, I'll hold off on any changes in this PR and try to consolidate the changes from here to PR #3216. and then we can close this PR.

Reply:
Per [your comment](https://github.com/distributed-system-analysis/pbench/pull/3216#issuecomment-1409466537) in #3216, I think you can close this PR, now.